### PR TITLE
LC-417 Adds github action config to deploy to sonatype.

### PR DESCRIPTION
**Pull request by kiratraynor**

### CODE REVIEW

**Review by kwatters — APPROVED** [review]
looks good to me.

File: .github/workflows/release-to-sonatype.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        required: true
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Maven Central Repository
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
```
Comment (rseitz) [inline .github/workflows/release-to-sonatype.yml:18]: in .github/workflows/maven.yaml we specify java-version 17
should we use 17 here too?

Comment (kwatters) [reply .github/workflows/release-to-sonatype.yml:18]: my preference is that we keep with java 11 compliance for now.. let's create another issue to track updating to a newer version of java.

Comment (meghanboyd) [reply .github/workflows/release-to-sonatype.yml:18]: Created LC-442 to track the task of evaluating Lucille for Java 17 upgrade.

**Review by rseitz — APPROVED** [review]

**Pull request merged**